src/reducers/errors.tsx

```diff
@@ -0,0 +1,53 @@
+import { Reducer } from 'redux';
+import { ActionType, createAction, getType } from 'typesafe-actions';
+
+export type ApplicationError = {
+  id: number;
+  message: string;
+};
+
+export type ErrorsState = {
+  errors: ApplicationError[];
+  nextErrorId: number;
+};
+
+export const initialState: ErrorsState = {
+  errors: [],
+  // This is a sequence to give each error a unique ID.
+  nextErrorId: 1,
+};
+
+export const actions = {
+  showError: createAction('SHOW_ERROR', (resolve) => {
```

Comment:
I wonder if `showError` is the best name, or if it should be something like `recordError`? I suppose with this implementation `showError` is accurate, as it will cause the error to appear, but if/when we enhance it to only show errors that belong to a particular component, then `showError` may no longer be accurate.

Reply:
yep, I am open to anything else. Could also be `add`/`remove`, I don't know. I like the high level naming but maybe we should have something more generic.